### PR TITLE
move library build to dist build

### PR DESCRIPTION
**Pull request by edewit**
towards: #34393
Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

### CODE REVIEW

File: js/apps/account-ui/pom.xml [hunk 1]
```diff
@@ -49,6 +49,31 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>jboss-release</id>
```
Comment (jonkoops) [inline js/apps/account-ui/pom.xml:53]: Is this the correct profile, or should it be `distribution`? Not actually sure what `jboss-release` is used for.

**Comment by drichtarik** [comment]
I remember @pskopek knowing what use we had for this profile

**Comment by ASzc** [comment]
We enable both `-Pdistribution -Pjboss-release` in the product build, so either works from our perspective. The content of this PR appears ok, although the linked issue discusses things that are more concerning for me. Trying to speed up a maven build can easily lead to a place where the maven build is made fragile, and breaks in a product scenario. Particularly I'm thinking about the javadocs, and how difficult that was to re-enable for product due to maven-rules-breaking overly aggressive optimization. However, in this context, if it works in the same way as it did before with the profiles enabled, that should be ok.

**Comment by drichtarik** [comment]
But it's needed for downstream release purposes both in keycloak-parent and keycloak-client as they are both based on the jboss-parent pom: https://github.com/jboss/jboss-parent-pom/blob/main/pom.xml

Using the profile ensures product artifacts are uploaded and deployed correctly

**Review by jonkoops — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**